app/scripts/keyring-controller.js

```diff
@@ -273,6 +272,7 @@ module.exports = class KeyringController extends EventEmitter {
   setSelectedAccount (address) {
     var addr = normalize(address)
     this.configManager.setSelectedAccount(addr)
+    this.emit('update')
```

Comment:
This is also its own bug fix, so *ideally* this would get its own PR, so that if the reviewer didn't like how it's implemented, it doesn't hold up the rest of your changes here.